### PR TITLE
[docs] updated crates/aptos-crypto/README.md

### PR DESCRIPTION
**Pull request by alinush**
### Description

Brings our cryptography crate's documentation in line with its **current, actual** implementation

### CODE REVIEW

File: crates/aptos-crypto/README.md [hunk 1]
```diff
@@ -4,32 +4,53 @@ title: Crypto
 custom_edit_url: https://github.com/aptos-labs/aptos-core/edit/main/crypto/crypto/README.md
 ---
 
-The crypto component hosts all the implementations of cryptographic primitives we use in Aptos: hashing, signing, and key derivation/generation. The parts of the library using traits.rs contains the crypto API enforcing type safety, verifiable random functions, EdDSA & MultiEdDSA signatures.
+The crypto component hosts all the implementations of cryptographic primitives we use in Aptos: hashing, (multi)signatures, and key derivation/generation.
+
+To enforce type safety for signature schemes, we rely on traits from  [`traits.rs`](src/traits.rs) and [`validatable.rs`](src/validatable.rs).
```
Comment (davidiw) [inline crates/aptos-crypto/README.md:9]:
```suggestion
To enforce type-safety for signature schemes, we rely on traits from  [`traits.rs`](src/traits.rs) and [`validatable.rs`](src/validatable.rs).
```

File: crates/aptos-crypto/README.md [hunk 2]
```diff
@@ -4,32 +4,53 @@ title: Crypto
...
+  + Implemented on top of Barreto-Scott-Lynn BLS12-381 elliptic curves
+- The **[Noise Protocol Framework](http://www.noiseprotocol.org/)**
+  - Used to create authenticated and encrypted communications channels between validators
+- **X25519** key exchange
+  + Based on the [x25519-dalek](https://docs.rs/x25519-dalek) crate
+  + Used in our implementation of the [Noise Protocol Framework](http://www.noiseprotocol.org/)
+
+## Traits for safer cryptography implementation
+
+Before implementing a cryptographic primitive, be sure to read [`traits.rs`](src/traits.rs) and [`validatable.rs`](src/validatable.rs) to understand how to comply with our API as well as **some** of the security issues involved.
```
Comment (davidiw) [inline crates/aptos-crypto/README.md:34]:
```suggestion
Before implementing a cryptographic primitive, be sure to read [`traits.rs`](src/traits.rs) and [`validatable.rs`](src/validatable.rs) to understand how to comply with our API as well as **some** of the security concerns involved.
```

File: crates/aptos-crypto/README.md [hunk 3]
```diff
@@ -4,32 +4,53 @@ title: Crypto
...
+    ├── traits.rs           # Traits for safer implementations of signature schemes
+    ├── validatable.rs      # Traits for deferring validation of group elements (e.g., public keys, signatures)
+    └── x25519.rs           # X25519 implementation
+
 ```
 
-Note: This crate historically had support for BLS12381, ECVRF, and SlIP-0010, though were removed due to lack of use. The last git revision before there removal is 00301524.
+## Changelog
+
+ - This crate historically had support for (a different) BLS12-381, [EC-VRF](https://tools.ietf.org/id/draft-goldbe-vrf-01.html#rfc.section.5), and [SLIP-0010](https://github.com/satoshilabs/slips/blob/master/slip-0010.md), though were removed due to lack of use. The last git revision before the removal is 00301524.
```
Comment (davidiw) [inline crates/aptos-crypto/README.md:56]: Let's delete this.

**Review by davidiw — APPROVED** [review]

File: crates/aptos-crypto/README.md (hunk 3 above)
Comment (alinush) [reply crates/aptos-crypto/README.md:56]: Why? It would be a good reminder in case we need to take a look back.

**Pull request closed without merging**